core/Map.carp

```diff
@@ -632,6 +632,8 @@
     (doc init "creates a dynamic pair, i.e. a list with two elements.")
     (defndynamic init [k v] (list k v)))
 
+  (deftype-dynamic Map (l))
```

Comment:
This call to `(l)` is a bit cryptic to me...

Reply:
This is not a call, it just says that `Map` has one member, `l`, which  is the underlying list structure!

Reply:
Maybe we should change the syntax to an array?

Reply:
Ah, I see. Yeah that would be better I think. Also give the member a more descriptive name?

Reply:
I changed it so that it doesn’t matter, but I suggest we make array the convention.

I also changed the member name to `buckets`.